lib/phase/permutation.js

```diff
@@ -64,6 +64,10 @@ module.exports = async validatedDoc => {
             job.freezeWindows = doc.jobs[jobName].freezeWindows;
         }
 
+        if (doc.jobs[jobName].stage !== undefined) {
```

Comment:
Why do we need reference to a `stage` from a `job` and persist that in DB?

Reply:
this wouldn't be persisted in the DB
just passed from config-parser to models temporarily

Reply:
It is getting persisted in DB for all the stage jobs.

Sample pipeline
https://github.com/sagar1312/screwdriver-config/blob/main/stage-with-parallel-jobs/screwdriver.yaml

Here are the entries from `jobs` table for this pipeline config
```
+-----------------+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
|name             |permutations                                                                                                                                                                                                                                                      |
+-----------------+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
|main             |[{"image":"node:12","commands":[{"name":"config-parse-error","command":"echo 'ValidationError: \"jobs.prod-green-deploy.prod-green-test\" is not allowed. \"jobs.prod-green-deploy.prod-green-certify\" is not allowed'; exit 1"}],"secrets":[],"environment":{}}]|
|component        |[{"annotations":{},"commands":[{"name":"init","command":"echo 'init'"}],"environment":{},"image":"node:14","secrets":[],"settings":{},"requires":["~pr","~commit"]}]                                                                                              |
|ci-deploy        |[{"annotations":{},"commands":[{"name":"init","command":"echo 'init'"}],"environment":{},"image":"node:14","secrets":[],"settings":{},"requires":["publish"],"stage":{"name":"integration"}}]                                                                     |
|publish          |[{"annotations":{},"commands":[{"name":"init","command":"echo 'init'"}],"environment":{},"image":"node:14","secrets":[],"settings":{},"requires":["component"]}]                                                                                                  |
|ci-certify       |[{"annotations":{},"commands":[{"name":"init","command":"echo 'init'"}],"environment":{},"image":"node:14","secrets":[],"settings":{},"requires":["ci-test"],"stage":{"name":"integration"}}]                                                                     |
|ci-test          |[{"annotations":{},"commands":[{"name":"init","command":"echo 'init'"}],"environment":{},"image":"node:14","secrets":[],"settings":{},"requires":["ci-deploy"],"stage":{"name":"integration"}}]                                                                   |
|prod-blue-deploy |[{"annotations":{},"commands":[{"name":"init","command":"echo 'init'"}],"environment":{},"image":"node:14","secrets":[],"settings":{},"requires":["ci-certify"],"stage":{"name":"production-blue"}}]                                                              |
|prod-blue-test   |[{"annotations":{},"commands":[{"name":"init","command":"echo 'init'"}],"environment":{},"image":"node:14","secrets":[],"settings":{},"requires":["prod-blue-deploy"],"stage":{"name":"production-blue"}}]                                                        |
|prod-blue-certify|[{"annotations":{},"commands":[{"name":"init","command":"echo 'init'"}],"environment":{},"image":"node:14","secrets":[],"settings":{},"requires":["prod-blue-test"],"stage":{"name":"production-blue"}}]                                                          |
|prod-green-deploy|[{"annotations":{},"commands":[{"name":"init","command":"echo 'init'"}],"environment":{},"image":"node:14","secrets":[],"settings":{},"requires":["ci-certify"],"stage":{"name":"production-green"}}]                                                             |
+-----------------+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+

```

Reply:
ah I meant as its own field
but yeah looks like it persists in permutations blob

I can probably remove the startFrom object but the stage name tag is nice to have for quick lookup

Reply:
Field `stage` stored under `permutations` in DB does not align with the schema
https://github.com/screwdriver-cd/data-schema/pull/491/files#r1312129998

The schema needs to be fixed so that a job can only point to a single stage.

Reply:
https://github.com/screwdriver-cd/data-schema/pull/529